app/controllers/main/routes.py

```diff
@@ -15,14 +15,16 @@ def home():
 
 @main_bp.route('/profile/<username>', methods = ['GET'])
 def profilePage(username):
-    upcomingEvents = getUpcomingEventsForUser(g.current_user)
+    if username != g.current_user.username or g.current_user.isCeltsAdmin or g.current_user.isCeltsStudentStaff:
+        return "", 500 #TODO: Link to an error page
+    upcomingEvents = getUpcomingEventsForUser(username)
     programs = Program.select()
     interests = Interest.select().where(Interest.user == g.current_user)
     interests_ids = [interest.program for interest in interests]
     if username == g.current_user.username or g.current_user.isCeltsAdmin or g.current_user.isCeltsStudentStaff:
         return render_template('/volunteer/volunteerProfile.html',
                                title="Volunteer Interest",
-                               user = g.current_user,
+                               user = username,
```

Comment:
you were passing a User object, and now you are passing a text username. Does it still work?

Reply:
I'm guessing no, since your template calls `user.username`, which doesn't work if user is just a string of text. Gotta test after you make changes.